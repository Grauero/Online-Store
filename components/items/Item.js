import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { Query } from 'react-apollo';

import DeleteItem from './DeleteItem';
import AddToCart from '../cart/AddToCart';
import ItemStyles from '../styles/ItemStyles';
import PriceTag from '../styles/PriceTag';
import Title from '../styles/Title';
import formatMoney from '../../util/formatMoney';
import { CURRENT_USER_QUERY } from '../../mutations/auth';

const Item = ({ item }) => (
  <Query query={CURRENT_USER_QUERY}>
    {({ data: { me } }) => {
      const editButton = item.user.id === me.id ? (
        <Link
          href={{
            pathname: 'update',
            query: { id: item.id }
          }}
        >
          <a>Edit &#9999;&#65039; </a>
        </Link>
      ) : null;

      const deleteButton = item.user.id === me.id ? <DeleteItem id={item.id}>Delete This Item</DeleteItem> : null;

      return (
        <ItemStyles>
          {item.image && <img src={item.image} alt={item.title} />}
          <Title>
            <Link
              href={{
                pathname: '/item',
                query: { id: item.id }
              }}
            >
              <a>{item.title}</a>
            </Link>
          </Title>
          <PriceTag>{formatMoney(item.price)}</PriceTag>
          <p>{item.description}</p>
          <div className="buttonList">
            {editButton}
            <AddToCart id={item.id} />
            {deleteButton}
          </div>
        </ItemStyles>
      );
    }}
  </Query>
);

Item.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired
};

export default Item;
