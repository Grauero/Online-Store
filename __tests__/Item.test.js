import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Item from '../components/items/Item';

const testItem = {
  id: 'id',
  title: 'title',
  price: 4000,
  description: 'description',
  image: 'image',
  largeImage: 'largeImage.'
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={testItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<Item item={testItem} />);
    const img = wrapper.find('img');

    expect(img.props().src).toBe(testItem.image);
    expect(img.props().alt).toBe(testItem.title);
  });

  it('renders the pricetag and title', () => {
    const wrapper = shallow(<Item item={testItem} />);
    const PriceTag = wrapper.find('PriceTag');

    expect(PriceTag.children().text()).toBe('$40');
    expect(wrapper.find('Title a').text()).toBe(testItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<Item item={testItem} />);
    const buttonList = wrapper.find('.buttonList');

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
