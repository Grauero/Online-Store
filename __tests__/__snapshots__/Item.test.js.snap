// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="title"
    src="image"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "id",
          },
        }
      }
    >
      <a>
        title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "id",
          },
        }
      }
    >
      <a>
        Edit ✏️ 
      </a>
    </Link>
    <AddToCart
      id="id"
    />
    <DeleteItem
      id="id"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
